dask/dataframe/io/io.py

```diff
@@ -213,6 +232,13 @@ def from_pandas(data, npartitions=None, chunksize=None, sort=True, name=None):
     return new_dd_object(dsk, name, data, divisions)
 
 
+def _infer_npartitions(data, chunksize):
+    """ Return the number of partitions based on an estimate of data size and the provided chunksize"""
+    partition_size = parse_bytes(chunksize)
+    mem_usage = total_mem_usage(data)
+    return int(ceil(mem_usage / partition_size))
```

Comment:
I would prefer that we just include these lines of code in the condition above.  I would prefer that we not make a new function for this.  See https://matthewrocklin.com/blog/work/2019/06/23/avoid-indirection for reasoning

Reply:
I get that this is used twice, but even then, I think I'd still prefer to see it inlined if that's ok with you.

Reply:
I believe `infer_npartitions` can be removed now.